### PR TITLE
Feature sms web detail view

### PR DESCRIPTION
**Pull request by dongilbert**
Please answer the following questions. 

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | N/A
| Related developer documentation PR URL | N/A
| Issues addressed (#s or URLs) | N/A
| BC breaks? | N
| Deprecations? | N

#### Description:
This very simply adds detail view for Text Messages and Web Notifications.

#### Steps to test this PR:
1. Have a Web Notification and Text Message with successful sends.
2. In the list view for the items, click the item title in the table to load the detail view.
3. Ensure the "Clicks" and "Contacts" tabs are properly populated.

### CODE REVIEW

**Comment by dbhurley** [comment]
This seems to touch a lot of code (83 files) and more than just add a view to SMS or Web Push. I only comment on that aspect because this PR breaks my local install. I'm still digging in to try and determine where the offending code lies. I'll add a second comment if I can trace the error.

**Comment by alanhartless** [comment]
Everything loaded was viewable for me. 

Couple minor things:
1. mautic.sms.notice.batch_deleted and mautic.notification.notice.batch_deleted strings arne't defined.
2. Thoughts on adding a New button to the list pages now that we have details pages as well? In theory, :-) should be as simple as https://github.com/mautic/mautic/blob/aa21004da6c31fcd57a3777ac9db5b1812c50ff9/app/bundles/EmailBundle/Views/Email/index.html.php#L14 with the appropriate permissions and route base since we already have the form.

**Comment by dongilbert** [comment]
@dbhurley This touches so many files due to `php-cs-fixer` fixing the code style.

File: app/bundles/NotificationBundle/Controller/NotificationController.php [hunk 1]
```diff
@@ -803,21 +809,113 @@ public function previewAction($objectId)
...
+                'page' => $page,
+                'items' => $contacts['results'],
+                'totalItems' => $contacts['count'],
+                'tmpl' => 'notificationcontacts',
+                'indexMode' => 'grid',
+                'link' => 'mautic_notification_contacts',
+                'sessionVar' => 'notification.contact',
+                'limit' => $limit,
+                'objectId' => $objectId,
+                'noContactList' => $emailRepo->getDoNotEmailList(),
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Controller/NotificationController.php]: Do we want no contact list for the notification channel?  I think this just returns email DNC right?

File: app/bundles/NotificationBundle/Entity/StatRepository.php [hunk 2]
```diff
@@ -120,19 +118,19 @@ public function getReadCount($emailIds = null, $listId = null)
         $q = $this->_em->getConnection()->createQueryBuilder();
 
         $q->select('count(s.id) as read_count')
-            ->from(MAUTIC_TABLE_PREFIX . 'push_notification_stats', 's');
+            ->from(MAUTIC_TABLE_PREFIX.'push_notification_stats', 's');
 
         if ($emailIds) {
             if (!is_array($emailIds)) {
-                $emailIds = array((int) $emailIds);
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Entity/StatRepository.php]: Let's name `$emailIds` to something applicable to Notifications

File: app/bundles/NotificationBundle/Entity/StatRepository.php [hunk 3]
```diff
@@ -150,11 +148,11 @@ public function getReadCount($emailIds = null, $listId = null)
      */
     public function getClickedRates($notificationIds, \DateTime $fromDate = null)
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Entity/StatRepository.php]: Is this method used?

File: app/bundles/NotificationBundle/Entity/StatRepository.php [hunk 4]
```diff
@@ -284,21 +284,22 @@ public function getMostNotifications($query, $limit = 10, $offset = 0)
...
     /**
-     * Get sent counts based grouped by email Id
+     * Get sent counts based grouped by email Id.
      *
      * @param array $emailIds
      *
      * @return array
      */
-    public function getSentCounts($emailIds = array(), \DateTime $fromDate = null)
+    public function getSentCounts($emailIds = [], \DateTime $fromDate = null)
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Entity/StatRepository.php]: Is this method used?

File: app/bundles/NotificationBundle/Event/NotificationSendEvent.php [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
 
 use Mautic\CoreBundle\Event\CommonEvent;
 use Mautic\EmailBundle\Entity\Email;
 use Mautic\LeadBundle\Entity\Lead;
 
 /**
- * Class NotificationSendEvent
- *
- * @package Mautic\NotificationBundle\Event
+ * Class NotificationSendEvent.
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Event/NotificationSendEvent.php]: This should probably be cleaned up a bit to remove unused methods, docblocks, etc from copying over from the EmailBundle

File: app/bundles/NotificationBundle/EventListener/NotificationSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+    /**
+     * @param NotificationSendEvent $event
+     */
+    public function onSend(NotificationSendEvent $event)
+    {
+        $content = $event->getContent();
+        $tokens = [];
+        /** @var \Mautic\SmsBundle\Api\AbstractSmsApi $smsApi */
+        $smsApi = $this->factory->getKernel()->getContainer()->get('mautic.sms.api');
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/EventListener/NotificationSubscriber.php]: This doesn't seem to be used.

File: app/bundles/NotificationBundle/Translations/en_US/messages.ini [hunk 7]
```diff
@@ -76,6 +76,8 @@ mautic.notification.timeline.status.failed="Failed"
 mautic.notification.config.form.notification.safari_web_id="Safari Web ID"
 mautic.notification.config.form.notification.safari_web_id.tooltip="Safari Web ID from OneSignal if you want to support Apple Push Notifications"
 
-mautic.notification.campaign.failed.not_contactable="Contact is not contactable on the Web Notification channel."
-mautic.notification.campaign.failed.not_subscribed="The contact has not subscribed to the Web Notification channel."
+mautic.notification.campaign.failed.not_contactable="Lead is not contactable on the Web Notification channel."
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Translations/en_US/messages.ini]: Probably didn't mean to revert these back to `lead`

File: app/bundles/NotificationBundle/Views/Notification/details.html.php [hunk 8]
```diff
@@ -41,12 +61,30 @@
...
+            <!-- some stats -->
+            <div class="pa-md">
+                <div class="row">
+                    <div class="col-sm-12">
+                        <div class="panel">
+                            <div class="panel-body box-layout">
+                                <div class="col-md-3 va-m">
+                                    <h5 class="text-white dark-md fw-sb mb-xs">
+                                        <span class="fa fa-line-chart"></span>
+                                        <?php echo $view['translator']->trans('mautic.sms.sent'); ?>
```
Comment (alanhartless) [inline app/bundles/NotificationBundle/Views/Notification/details.html.php]: Is this the right language string?

File: app/bundles/NotificationBundle/Translations/en_US/messages.ini (hunk 7 above)
Comment (dongilbert) [inline app/bundles/NotificationBundle/Translations/en_US/messages.ini]: Merge failure

**Comment by alanhartless** [comment]
https://github.com/dongilbert/mautic/pull/12

**Comment by mqueme** [comment]
I get an unexpected token javascript error
screen shot 2016-08-10 at 14 53 23

**Comment by escopecz** [comment]
@mqueme that's because there is some PHP error in the ajax response and it is written in HTML and the JSON response is bellow. What is the PHP error?

**Comment by dbhurley** [comment]
+1

**Pull request merged**